packages/wrangler/src/api/integrations/bindings/services.ts

```diff
@@ -0,0 +1,108 @@
+import { Response } from "miniflare";
+import { fetch, type Request } from "undici";
+import { getRegisteredWorkers } from "../../../dev-registry";
+import type { WorkerDefinition, WorkerRegistry } from "../../../dev-registry";
+import type { RequestInit } from "undici";
+
+/**
+ * Gets service bindings that proxy requests to external workers running locally, so that they
+ * can be passed to miniflare and used to fetch from such workers by the next app.
+ *
+ * @param services the services requested by the user
+ * @returns the service bindings ready to be passed to miniflare
+ */
+export async function getServiceBindings(
+	services:
+		| {
+				binding: string;
+				service: string;
+		  }[]
+		| undefined = []
+): Promise<ServiceBindings | undefined> {
+	if (services.length === 0) {
+		return;
+	}
+
+	let registeredWorkers: WorkerRegistry | undefined;
+
+	try {
+		registeredWorkers = await getRegisteredWorkers();
+	} catch {
+		/* */
+	}
+
+	if (!registeredWorkers) {
+		return;
+	}
+
+	const [foundServices, _missingServices] = services.reduce(
+		([found, missing], { binding: bindingName, service: serviceName }) => {
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const service = registeredWorkers![serviceName];
+			return service
+				? [
+						[...found, { bindingName, serviceName, workerDefinition: service }],
+						missing,
+				  ]
+				: [found, [...missing, { bindingName }]];
+		},
+		[[], []] as [
+			AvailableBindingInfo[],
+			Pick<AvailableBindingInfo, "bindingName">[]
+		]
+	);
```

Comment:
You do love your reduces 😄 

```suggestion
  const foundServices: AvailableBindingInfo[] = [];
  const _missingServices: Pick<AvailableBindingInfo, "bindingName">[] = [];
  for (const {binding: bindingName, service: serviceName} of services) {
     const worker = registeredWorkers[serviceName]
     if (worker) {
       foundServices.push({ bindingName, serviceName, workerDefinition: worker });
     } else {
       _missingServices.push({bindingName});
  }
```